### PR TITLE
Various docs updates

### PR DESCRIPTION
**Pull request by Zac-HD**
The commits should all speak for themselves - this is really a collection of smaller things.

- I really like Sphinx cross-references.  I especially think they're important in the changelog, so readers can immediately jump to more detail when a new thing is introduced.  I wrote a small docs guide mostly to point out how nice they are 😜 

- I dislike deprecated code examples, outdated references, and defaulting to Python 2.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -67,7 +67,7 @@
 todo_include_todos = False
 
 intersphinx_mapping = {
-    'python': ('http://docs.python.org/', None),
+    'python': ('http://docs.python.org/3/', None),
```
Comment (DRMacIver) [inline docs/conf.py]: What does this actually do?

File: guides/documentation.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+That means every contributor gets to write their changelog!
+
+A changelog entry should:
+
+- concisely describe what changed and why
+- use Sphinx cross-references to any functions or classes mentioned
+- include the current date and proposed version number (also updated in src/hypothesis/version.py)
+- if closing an issue, mention it with the issue role to generate a link
+- if a first-time contribution, optionally write a thanks by name
+  (for publication by the maintainers).  Also add yourself to contributors.rst!
```
Comment (DRMacIver) [inline guides/documentation.rst]: I'm not sure thanks should just be for first-time contributors. It'd be nice to people in general more credit for features than we currently do. Maybe we should just start adopting a policy of adding credit to all releases?

File: guides/documentation.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
+=====================================
+The Hypothesis Documentation Handbook
```
Comment (DRMacIver) [inline guides/documentation.rst]: 👍 to this. It should probably be referenced from CONTRIBUTING.rst though.

File: docs/conf.py (hunk 1 above)
Comment (DRMacIver) [inline docs/conf.py]: Oh, I see, it means that when we ref a Python standard library thing it goes to the Python 3 docs. That's definitely a good change.

File: guides/documentation.rst (hunk 3 above)
Comment (Zac-HD) [inline guides/documentation.rst]: Done 😄

**Comment by Zac-HD** [comment]
Ping @HypothesisWorks/hypothesis-python-contributors; ready to merge.

**Comment by Zac-HD** [comment]
@DRMacIver, ready to merge.

**Pull request merged**